asf_core_data/pipeline/data_joining/merge_proc_datasets.py

```diff
@@ -0,0 +1,206 @@
+        "No HP": "No HP",
+        "air source heat pump": "Air Source Heat Pump",
+        "ground source heat pump": "Ground/Water Source Heat Pump",
+        "heat pump": "Undefined or Other Heat Pump Type",
+        "water source heat pump": "Ground/Water Source Heat Pump",
+        "community heat pump": "Undefined or Other Heat Pump Type",
+        "Exhaust Air Heat Pump": "Undefined or Other Heat Pump Type",
+        "Air Source Heat Pump": "Air Source Heat Pump",
+        "Ground/Water Source Heat Pump": "Ground/Water Source Heat Pump",
+    }
+
+    # Overwrite with tech type from MCS if MCS data exists
+    epc_mcs_df["HP_TYPE"] = np.where(
+        epc_mcs_df["MCS_AVAILABLE"], epc_mcs_df["tech_type"], epc_mcs_df["HP_TYPE"]
+    )
+
+    epc_mcs_df["HP_TYPE"] = epc_mcs_df["HP_TYPE"].map(type_dict)
+
+    return epc_mcs_df
+
+
+def add_mcs_installer_data(epc_mcs_installations):
+    """Add MCS installer data to joined EPC and MCS installations dataframe based on installer ID.
+
+    Args:
+        epc_mcs_installations (pd.DataFrame): EPC and MCS installations data.
+    """
+
+    newest_hist_inst_batch = data_batches.get_latest_hist_installers()
+
+    print(newest_hist_inst_batch)
+
+    # # Load MCS
+    mcs_inst_data = data_getters.load_s3_data(
+        base_config.BUCKET_NAME,
+        newest_hist_inst_batch,
+    )
+
+    mcs_inst_data.rename(columns={"postcode": "POSTCODE"}, inplace=True)
+
+    # TODO Code to merge data with epc_mcs_installations based on company_unique_id
+    # It should be an outer merge keeping, keeping installer data with no matches to MCS installations
+    # Their MCS installations and EPC fields would be NaN
+
+
+def merging_pipeline():
+    """Merge EPC and MCS installation and installer data to create gold schema like dataset.
+    - Load EPC data
+    - Get best approximation for installation date
+    - Merge with MCS installations and reformatting
+    - Merge with MCS installers
+    - Reformat postcode
+    - Save output to S3
+
+    """
+
+    # Load the processed EPC data
+    prep_epc = load_preprocessed_epc_data(
+        data_path="S3", version="preprocessed", batch="newest"
+    )
+
+    # Add more precise estimations for heat pump installation dates via MCS data
+    epc_with_MCS_dates = merge_install_dates.manage_hp_install_dates(prep_epc)
+
+    # Merge EPC with MCS installations
+    epc_mcs_insts = merge_proc_epc_and_mcs_installations(epc_with_MCS_dates)
+
+    # Merge with MCS installers
+    epc_mcs_complete = add_mcs_installer_data(epc_mcs_insts)
+
+    # Reformat postcode field to include no space
+    epc_mcs_complete = data_cleaning.reformat_postcode(
+        epc_mcs_complete, postcode_var_name="POSTCODE", white_space="remove"
+    )
+
+    # Add geographical data
+
+    # Save final merged dataset
+    data_getters.save_to_s3(
+        bucket_name="asf-core-data",
+        output_var=epc_mcs_complete,
+        output_file_path='"output/mcs/gold/merged_epc_mcs.csv"',
+    )
```

Comment:
I think we are missing something like:

```suggestion
    )
    if __name__ == "__main__":
       merging_pipeline()
```
in order to run the script from the command line. Or maybe this function is called somewhere else and I haven't seen that.

Reply:
Nope, doesn't exist yet. Added something simple for now, can be improved later on.